src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentProperty.java

```diff
@@ -154,51 +155,79 @@ private void initDateConverter() {
 
-			if (dateFormat == DateFormat.none) {
+			if (dateFormats.length == 0) {
 				LOGGER.warn(
 						String.format("No DateFormat defined for property %s. Make sure you have a Converter registered for %s",
 								property, actualType.getSimpleName()));
 				return;
 			}
 
-			ElasticsearchDateConverter converter = null;
+			List<ElasticsearchDateConverter> converters = new ArrayList<>();
 
-			if (dateFormat == DateFormat.custom) {
-				String pattern = field.pattern();
+			for (int i = 0; i < dateFormats.length; i++) {
+				DateFormat dateFormat = dateFormats[i];
 
-				if (!StringUtils.hasLength(pattern)) {
-					throw new MappingException(
-							String.format("Property %s is annotated with FieldType.%s and a custom format but has no pattern defined",
-									property, field.type().name()));
-				}
+				if (dateFormat == DateFormat.custom) {
+					String pattern = field.pattern();
+
+					if (!StringUtils.hasLength(pattern)) {
+						throw new MappingException(String.format(
+								"Property %s is annotated with FieldType.%s and a custom format but has no pattern defined", property,
+								field.type().name()));
+					}
 
-				converter = ElasticsearchDateConverter.of(pattern);
-			} else {
-
-				switch (dateFormat) {
-					case weekyear:
-					case weekyear_week:
-					case weekyear_week_day:
-						LOGGER.warn("no Converter available for " + actualType.getName() + " and date format " + dateFormat.name()
-								+ ". Use a custom converter instead");
-						break;
-					default:
-						converter = ElasticsearchDateConverter.of(dateFormat);
-						break;
+					converters.add(ElasticsearchDateConverter.of(pattern));
+				} else {
+
+					switch (dateFormat) {
+						case weekyear:
+						case weekyear_week:
+						case weekyear_week_day:
+							LOGGER.warn("no Converter available for " + actualType.getName() + " and date format " + dateFormat.name()
+									+ ". Use a custom converter instead");
+							break;
+						default:
+							converters.add(ElasticsearchDateConverter.of(dateFormat));
+							break;
+					}
 				}
+
 			}
 
-			if (converter != null) {
-				ElasticsearchDateConverter finalConverter = converter;
+			if (!converters.isEmpty()) {
 				propertyConverter = new ElasticsearchPersistentPropertyConverter() {
-					final ElasticsearchDateConverter dateConverter = finalConverter;
+					final List<ElasticsearchDateConverter> dateConverters = converters;
+
+					@SuppressWarnings("unchecked")
+					@Override
+					public Object read(String s) {
+						if (isTemporalAccessor) {
+							for (ElasticsearchDateConverter dateConverter : dateConverters) {
+								try {
+									return dateConverter.parse(s, (Class<? extends TemporalAccessor>) actualType);
+								} catch (Exception e) {
+								}
+							}
+						} else { // must be date
+							for (ElasticsearchDateConverter dateConverter : dateConverters) {
+								try {
+									return dateConverter.parse(s);
+								} catch (Exception e) {
+								}
```

Comment:
we should at least log the exception here

Reply:
If no converter can parse the date, we throw a `RuntimeException` at the very end.